### PR TITLE
bash script version watchdog

### PR DESCRIPTION
**Pull request by KagurazakaNyaa**
Try to use bash script to implement factorio version check
Before merging this PR, you need to create a token that can be written to this project at https://github.com/settings/personal-access-tokens/new and configure it into the REPO_TOKEN secret so that GitHub Action can trigger docker builds

ref #459

### CODE REVIEW

**Comment by Fank** [comment]
Added REPO_TOKEN

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Check Update
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    - cron: "0 0 * * *"
```
Comment (Fank) [inline .github/workflows/update.yml:8]: Currently the watchdog runs every 5 min (https://github.com/factoriotools/factorio-docker-watchdog/blob/master/factorio-docker-watchdog.go#L23) would it be possible increase the check interval?
Would every hour be enough? i think once per day is not enough.

File: update.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+fi
+rm -f -- "$tmpfile"
+
+git config user.name github-actions
+git config user.email github-actions@github.com
+git add buildinfo.json
+git commit -a -m "Auto Update Factorio to version: "$version
+git tag -f latest
+git push
+git push origin --tags -f
```
Comment (Fank) [inline update.sh:57]:
```suggestion
git push origin --tags -f

```

**Review by Fank — CHANGES_REQUESTED** [review]

File: .github/workflows/update.yml (hunk 1 above)
Comment (PHIDIAS0303) [reply .github/workflows/update.yml]: once an hour seems already enough, not really need 5 minutes

**Review by Fank — APPROVED** [review]

File: .github/workflows/update.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.REPO_TOKEN }}
+
+      # Runs a single command using the runners shell
```
Comment (SuperSandro2000) [inline .github/workflows/update.yml:28]:
```suggestion
jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v3
        with:
          token: ${{ secrets.REPO_TOKEN }}

```

File: .github/workflows/update.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+name: Check Update
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
```
Comment (SuperSandro2000) [inline .github/workflows/update.yml:7]:
```suggestion
on:
  schedule:
```

File: .github/workflows/update.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    - cron: "0 * * * *"
+  workflow_dispatch:
+    inputs:
+      desc:
+        description: "desc"
+        required: false
```
[the remark this answers is not in the document]
Comment (KagurazakaNyaa) [reply .github/workflows/update.yml:13]: Indeed this is the problem, maybe I can check if it's in the fork project by adding an if statement

File: update.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (KagurazakaNyaa) [reply update.sh:57]: If you directly push the latest tag, the existing tag cannot be correctly overwritten

**Comment by KagurazakaNyaa** [comment]
Is there more that needs to be changed, and if not maybe consider merging?

**Comment by Fank** [comment]
@KagurazakaNyaa Thanks for adding an update solution via github actions, it looks like a very good option which is also more transparent then my solution before.
The jobs are running for up to 20secs which looks good based on the free pipeline quota.
Sadly i am not that familiar with the github actions, so i didn't remeber about the schedule option.

Thanks again for this PR, i think that will make a lot of people happy.

**Pull request merged**